app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/RecentSyncedTabFeature.kt

```diff
@@ -95,12 +100,26 @@ class RecentSyncedTabFeature(
             .maxByOrNull { it.device.lastAccessTime ?: 0 }
             ?.let {
                 val tab = it.tabs.firstOrNull()?.active() ?: return
+
+                val currentTime = System.currentTimeMillis()
+                val maxAgeInMs = TimeUnit.DAYS.toMillis(DAYS_HISTORY_FOR_PREVIEW_IMAGE)
+                val history = historyStorage.getDetailedVisits(
+                    start = currentTime - maxAgeInMs,
+                    end = currentTime
+                )
+
+                // Searching history entries for any that share a top level domain and have a
+                // preview image URL available casts a wider net for finding a suitable image.
+                val imageUrl = history.find { entry ->
```

Comment:
Could we just name this `previewImageUrl`?  Generally, I prefer if the variable naming follows the named parameter that we are passing to if there's no obvious naming conflict.